BE/PoroDev.UserManagementService/Services/UserService.cs

```diff
@@ -363,6 +370,112 @@ public async Task<CommunicationModel<RegisterUserResponse>> RegisterUser(Registe
             return returnContext;
         }
 
-        
+        private async Task<CommunicationModel<DataUserModel>> CheckUserFieldsUpdate(UserUpdateRequestGatewayToService updateModel)
+        {
+            try
+            {
+                CheckFullName(updateModel.Name, updateModel.Lastname);
+                await CheckEmailUpdate(updateModel.Email);
+                CheckPassword(updateModel.PasswordUnhashed);
+                CheckPosition(updateModel.Position);
+            }
+            catch (EmailFormatException ex)
+            {
+
+                string exceptionType = nameof(EmailFormatException);
+                string humanReadableMessage = ex.HumanReadableErrorMessage;
+
+                var responseException = new CommunicationModel<DataUserModel>()
+                {
+                    Entity = null,
+                    ExceptionName = exceptionType,
+                    HumanReadableMessage = humanReadableMessage
+                };
+
+                return responseException;
+            }
+            catch (PasswordFormatException ex)
+            {
+                string exceptionType = nameof(PasswordFormatException);
+                string humanReadableMessage = ex.HumanReadableErrorMessage;
+
+                var responseException = new CommunicationModel<DataUserModel>()
+                {
+                    Entity = null,
+                    ExceptionName = exceptionType,
+                    HumanReadableMessage = humanReadableMessage
+                };
+
+                return responseException;
+            }
+            catch (FullNameFormatException ex)
+            {
+                string exceptionType = nameof(FullNameFormatException);
+                string humanReadableMessage = ex.HumanReadableErrorMessage;
+
+                var responseException = new CommunicationModel<DataUserModel>()
+                {
+                    Entity = null,
+                    ExceptionName = exceptionType,
+                    HumanReadableMessage = humanReadableMessage
+                };
+
+                return responseException;
+            }
+            catch (PositionFormatException ex)
+            {
+                string exceptionType = nameof(PositionFormatException);
+                string humanReadableMessage = ex.HumanReadableErrorMessage;
+
+                var responseException = new CommunicationModel<DataUserModel>()
+                {
+                    Entity = null,
+                    ExceptionName = exceptionType,
+                    HumanReadableMessage = humanReadableMessage
+                };
+
+                return responseException;
+            }
+
+            return new CommunicationModel<DataUserModel>() { Entity = null, ExceptionName = null, HumanReadableMessage = null };
+        }
+
+        private async Task CheckEmailUpdate(string email)
+        {
```

Comment:
There is a lot of duplicate code. Could you break down this into smaller functions?

Reply:
I try to break down some parts of the code. Review again and comment if there are some additional suggestions about that.